pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -134,12 +137,12 @@ func (h *handler) getArgsEnvAndStatus(infraObj *infraObject, args map[string]int
 
 	// cloud-init will split the hostname on '.' and set the hostname to the first chunk. This causes an issue where all
 	// nodes in a machine pool may have the same node name in Kubernetes. Converting the '.' to '-' here prevents this.
-	cmd = append(cmd, strings.ReplaceAll(infraObj.meta.GetName(), ".", "-"))
+	cmd = append(cmd, strings.ReplaceAll(machine.GetName(), ".", "-"))
```

Comment:
Why is this changing?

Reply:
mistake on my end, I reverted the change